### PR TITLE
Issue with http

### PR DESCRIPTION
**Pull request by NoMan2000**
Changed the http protocol to https for pecl.php.net.  Fixed an issue with updating because composer complained about non-secure access.  Updated the readme to reflect that some users might have to use sudo to get commands to work.

Could not do this for windows.php.net since it seems to be missing a server listening to that port, the connection hangs up if you try to curl https://windows.php.net  

Updated the doc blocks since it seems things are missing or incorrectly documented in a few places.

### CODE REVIEW

**Comment by NoMan2000** [comment]
@Hywan Can you review the pull request?

File: src/Base/Abstracts/Package.php [hunk 1]
```diff
@@ -53,6 +60,9 @@ public function getSimpleName()
         return $m[2];
     }
 
+    /**
```
Comment (jubianchi) [inline src/Base/Abstracts/Package.php]: why adding such comments ?

File: src/Package/PHP/Command/Release.php [hunk 2]
```diff
@@ -49,8 +53,8 @@ class Release implements Interfaces\Package\Release
      */
     protected $pkg = null;
 
-    /*
-     * @var Closure
+    /**
+     * @var null|\Closure
```
Comment (jubianchi) [inline src/Package/PHP/Command/Release.php]: `callable` instead of `\Closure` ?

File: src/Config.php [hunk 3]
```diff
@@ -4,10 +4,22 @@
...
 use Composer;
 
+/**
+ * Class Config
+ * @package Pickle
+ */
 class Config extends Composer\Config
 {
+    /**
+     *
```
Comment (Quix0r) [inline src/Config.php]: Please document more!

File: src/Base/Abstracts/Package.php [hunk 4]
```diff
@@ -73,6 +86,9 @@ public function getUniqueNameForFs()
         other engines. Lets see if this is needed to be
         moved into the Interface so each engines package
         is forced to implement it on its own. But so far ... */
+    /**
+     *
```
Comment (Quix0r) [inline src/Base/Abstracts/Package.php]: Please document more!

File: src/Base/Abstracts/Package.php [hunk 5]
```diff
@@ -64,6 +74,9 @@ public function getVendorName()
         return $m[1];
     }
 
+    /**
+     * @return string
```
Comment (Quix0r) [inline src/Base/Abstracts/Package.php]: Please document more!

File: src/Base/Abstracts/Package.php (hunk 1 above)
Comment (Quix0r) [inline src/Base/Abstracts/Package.php]: Adding them is good for documentation, but leaving them empty is not.

File: src/Base/Abstracts/Package.php [hunk 6]
```diff
@@ -40,8 +40,15 @@
...
 use Composer\Package\Version\VersionParser;
 
+/**
+ * Class Package
+ * @package Pickle\Base\Abstracts
+ */
 class Package extends CompletePackage
 {
+    /**
+     * @return string
```
Comment (Quix0r) [inline src/Base/Abstracts/Package.php]: Please document more!

File: src/Base/Abstracts/Package.php [hunk 7]
```diff
@@ -40,8 +40,15 @@
 use Pickle\Package\Util\Header;
 use Composer\Package\Version\VersionParser;
 
+/**
+ * Class Package
```
Comment (Quix0r) [inline src/Base/Abstracts/Package.php]: What is this class really for? Okay, it is a class, yeah, but what does it generally do?

File: src/Config.php [hunk 8]
```diff
@@ -4,10 +4,22 @@
 
 use Composer;
 
+/**
+ * Class Config
```
Comment (Quix0r) [inline src/Config.php]: Same for `Package` class, what is it for?

File: src/Console/Application.php [hunk 9]
```diff
@@ -5,18 +5,37 @@
 use Symfony\Component\Console\Application as BaseApplication;
 use Pickle\Console\Helper;
 
+/**
+ * Class Application
```
Comment (Quix0r) [inline src/Console/Application.php]: Same for `Package` class, what is it for?

File: src/Console/Application.php [hunk 10]
```diff
@@ -5,18 +5,37 @@
...
+ * @package Pickle\Console
+ */
 class Application extends BaseApplication
 {
+    /**
+     *
```
Comment (Quix0r) [inline src/Console/Application.php]: Please document more.

File: src/Console/Application.php [hunk 11]
```diff
@@ -5,18 +5,37 @@
...
+     */
     const NAME = 'pickle';
+
+    /**
+     *
```
Comment (Quix0r) [inline src/Console/Application.php]: Please document more.

File: src/Console/Application.php [hunk 12]
```diff
@@ -5,18 +5,37 @@
...
+     */
     public function __construct($name = null, $version = null)
     {
         self::checkExtensions();
 
         parent::__construct($name ?: static::NAME, $version ?: (static::VERSION === '@' . 'pickle-version@' ? 'source' : static::VERSION));
     }
 
+    /**
+     * @return \Symfony\Component\Console\Helper\HelperSet
```
Comment (Quix0r) [inline src/Console/Application.php]: Please document more.

File: src/Console/Application.php [hunk 13]
```diff
@@ -26,6 +45,9 @@ protected function getDefaultHelperSet()
         return $helperSet;
     }
 
+    /**
+     * @return array|\Symfony\Component\Console\Command\Command[]
```
Comment (Quix0r) [inline src/Console/Application.php]: Please document more.

File: src/Console/Application.php [hunk 14]
```diff
@@ -43,6 +65,9 @@ protected function getDefaultCommands()
         return $commands;
     }
 
+    /**
+     *
```
Comment (Quix0r) [inline src/Console/Application.php]: Please document more.

File: src/Console/Command/InstallerCommand.php [hunk 15]
```diff
@@ -113,6 +113,7 @@ protected function configure()
      * @param string          $path
      * @param InputInterface  $input
      * @param OutputInterface $output
+     * @throws \Exception
```
Comment (Quix0r) [inline src/Console/Command/InstallerCommand.php]: Thrown when what happened?

File: src/Console/Command/SelfUpdateCommand.php [hunk 16]
```diff
@@ -9,10 +9,20 @@
 use Humbug\SelfUpdate\Updater;
 use Humbug\SelfUpdate\Exception;
 
+/**
+ * Class SelfUpdateCommand
```
Comment (Quix0r) [inline src/Console/Command/SelfUpdateCommand.php]: Same as `Package`.

File: src/Console/Command/SelfUpdateCommand.php [hunk 17]
```diff
@@ -9,10 +9,20 @@
...
+ * @package Pickle\Console\Command
+ */
 class SelfUpdateCommand extends Command
 {
+    /**
+     *
```
Comment (Quix0r) [inline src/Console/Command/SelfUpdateCommand.php]: Please document more.

File: src/Console/Command/SelfUpdateCommand.php [hunk 18]
```diff
@@ -9,10 +9,20 @@
...
+     */
     const PHAR_NAME = 'pickle.phar';
 
+    /**
+     *
```
Comment (Quix0r) [inline src/Console/Command/SelfUpdateCommand.php]: Please document more.

File: src/Console/Command/SelfUpdateCommand.php [hunk 19]
```diff
@@ -27,6 +37,10 @@ protected function configure()
         ;
     }
 
+    /**
```
Comment (Quix0r) [inline src/Console/Command/SelfUpdateCommand.php]: Please document more.

File: src/Package/Convey.php [hunk 20]
```diff
@@ -41,10 +41,22 @@
 use Pickle\Package\Convey\Command\Type;
 use Pickle\Base\Util;
 
+/**
+ * Class Convey
```
Comment (Quix0r) [inline src/Package/Convey.php]: Same as with `Package`.

File: src/Package/Convey.php [hunk 21]
```diff
@@ -41,10 +41,22 @@
...
+ * @package Pickle\Package
+ */
 class Convey
 {
+    /**
```
Comment (Quix0r) [inline src/Package/Convey.php]: Please document more.

File: src/Package/Convey.php [hunk 22]
```diff
@@ -55,6 +67,11 @@ public function __construct($path, ConsoleIO $io)
         $this->command = Factory::getCommand($type, $path, $io);
     }
 
+    /**
```
Comment (Quix0r) [inline src/Package/Convey.php]: Please document more.

File: src/Package/PHP/Command/Release.php [hunk 23]
```diff
@@ -42,15 +42,19 @@
 use Pickle\Package\Util\Header;
 use Composer\Package\Version\VersionParser;
 
+/**
+ * Class Release
```
Comment (Quix0r) [inline src/Package/PHP/Command/Release.php]: Same as with `Package`.

File: src/Package/PHP/Command/Release.php [hunk 24]
```diff
@@ -73,6 +77,11 @@ public function __construct($path, $cb = null, $noConvert = false)
         $this->noConvert = $noConvert;
     }
 
+    /**
```
Comment (Quix0r) [inline src/Package/PHP/Command/Release.php]: Please document more.

File: src/Package/PHP/Command/Release.php [hunk 25]
```diff
@@ -155,6 +165,9 @@ public function create(array $args = array())
         }
     }
 
+    /**
+     *
```
Comment (Quix0r) [inline src/Package/PHP/Command/Release.php]: Please document more.

File: src/Package/PHP/Command/Release/Windows/Binary.php [hunk 26]
```diff
@@ -41,24 +41,28 @@
 use Pickle\Package\PHP\Util\PackageXml;
 use Pickle\Package\Util\Header;
 
+/**
+ * Class Binary
```
Comment (Quix0r) [inline src/Package/PHP/Command/Release/Windows/Binary.php]: Same as with `Package`.

File: src/Package/PHP/Command/Release/Windows/Binary.php [hunk 27]
```diff
@@ -41,24 +41,28 @@
...
 class Binary implements Interfaces\Package\Release
 {
     /**
      * @var \Pickle\Base\Interfaces\Package
      */
     protected $pkg = null;
 
-    /*
-     * @var Closure
+    /**
```
Comment (Quix0r) [inline src/Package/PHP/Command/Release/Windows/Binary.php]: Good one. :-)

File: src/Package/PHP/Command/Release/Windows/Binary.php [hunk 28]
```diff
@@ -77,11 +81,19 @@ public function __construct($path, $cb = null, $noConvert = false)
         $this->noConvert = $noConvert;
     }
 
+    /**
+     * @throws \Exception
```
Comment (Quix0r) [inline src/Package/PHP/Command/Release/Windows/Binary.php]: As on many places: Please document more.

File: src/Package/PHP/Command/Release/Windows/Binary.php [hunk 29]
```diff
@@ -77,11 +81,19 @@ public function __construct($path, $cb = null, $noConvert = false)
...
+     */
     public function __destruct()
     {
         $this->composerJsonBak($this->pkg, true);
     }
 
+    /**
```
Comment (Quix0r) [inline src/Package/PHP/Command/Release/Windows/Binary.php]: As on many places: Please document more.

File: src/Package/PHP/Command/Release/Windows/Binary.php [hunk 30]
```diff
@@ -102,6 +114,11 @@ protected function composerJsonBak(\Pickle\Base\Interfaces\Package $pkg, $restor
         }
     }
 
+    /**
```
Comment (Quix0r) [inline src/Package/PHP/Command/Release/Windows/Binary.php]: As on many places: Please document more.

File: src/Package/PHP/Command/Release/Windows/Binary.php [hunk 31]
```diff
@@ -147,6 +164,10 @@ protected function readPackage($path)
         return $package;
     }
 
+    /**
```
Comment (Quix0r) [inline src/Package/PHP/Command/Release/Windows/Binary.php]: As on many places: Please document more.

File: src/Package/PHP/Command/Release/Windows/Binary.php [hunk 32]
```diff
@@ -250,6 +272,10 @@ public function create(array $args = array())
         $zip->close();
     }
 
+    /**
```
Comment (Quix0r) [inline src/Package/PHP/Command/Release/Windows/Binary.php]: As on many places: Please document more.

File: src/Package/PHP/Package.php [hunk 33]
```diff
@@ -39,6 +39,10 @@
 use Pickle\Base\Abstracts;
 use Pickle\Base\Util\GitIgnore;
 
+/**
+ * Class Package
```
Comment (Quix0r) [inline src/Package/PHP/Package.php]: As on many places: Please document more.

File: src/Package/PHP/Package.php [hunk 34]
```diff
@@ -92,12 +96,16 @@ public function setRootDir($path)
         $this->path = $path;
     }
 
+    /**
```
Comment (Quix0r) [inline src/Package/PHP/Package.php]: As on many places: Please document more.

File: src/Package/PHP/Package.php [hunk 35]
```diff
@@ -92,12 +96,16 @@ public function setRootDir($path)
...
+     * @param $stability
+     */
     public function setStability($stability)
     {
         $this->stability = $stability;
     }
 
     /**
+     * @throws \Exception
```
Comment (Quix0r) [inline src/Package/PHP/Package.php]: As on many places: Please document more.

File: src/Package/PHP/Package.php [hunk 36]
```diff
@@ -130,6 +138,10 @@ public function getConfigureOptions()
         return $options;
     }
 
+    /**
```
Comment (Quix0r) [inline src/Package/PHP/Package.php]: As on many places: Please document more.

File: src/Package/PHP/Package.php [hunk 37]
```diff
@@ -281,6 +293,10 @@ public function getFiles()
         );
     }
 
+    /**
```
Comment (Quix0r) [inline src/Package/PHP/Package.php]: As on many places: Please document more.

File: src/Package/PHP/Package.php [hunk 38]
```diff
@@ -306,6 +322,10 @@ public function getVersionFromHeader()
         return [trim($version_define), $version];
     }
 
+    /**
```
Comment (Quix0r) [inline src/Package/PHP/Package.php]: As on many places: Please document more.

File: src/Package/PHP/Package.php [hunk 39]
```diff
@@ -317,6 +337,9 @@ protected function extConfigIsIn($path)
         }
     }
 
+    /**
```
Comment (Quix0r) [inline src/Package/PHP/Package.php]: As on many places: Please document more.

File: src/Package/PHP/Util/ConvertChangeLog.php [hunk 40]
```diff
@@ -36,9 +36,19 @@
 
 namespace Pickle\Package\PHP\Util;
 
+/**
+ * Class ConvertChangeLog
```
Comment (Quix0r) [inline src/Package/PHP/Util/ConvertChangeLog.php]: As on many places: Please document more.

File: src/Package/PHP/Util/ConvertChangeLog.php [hunk 41]
```diff
@@ -36,9 +36,19 @@
...
+ * @package Pickle\Package\PHP\Util
+ */
 class ConvertChangeLog
 {
+    /**
+     * @var string
```
Comment (Quix0r) [inline src/Package/PHP/Util/ConvertChangeLog.php]: As on many places: Please document more.

File: src/Package/PHP/Util/ConvertChangeLog.php [hunk 42]
```diff
@@ -36,9 +36,19 @@
...
+     */
     private $path;
+    /**
+     * @var
```
Comment (Quix0r) [inline src/Package/PHP/Util/ConvertChangeLog.php]: As on many places: Please document more.

File: src/Package/PHP/Util/ConvertChangeLog.php [hunk 43]
```diff
@@ -53,6 +63,9 @@ public function __construct($path)
         $this->path = $path;
     }
 
+    /**
+     *
```
Comment (Quix0r) [inline src/Package/PHP/Util/ConvertChangeLog.php]: As on many places: Please document more.

File: src/Package/PHP/Util/ConvertChangeLog.php [hunk 44]
```diff
@@ -76,6 +89,9 @@ public function parse()
         $this->changelog = $changelog;
     }
 
+    /**
+     *
```
Comment (Quix0r) [inline src/Package/PHP/Util/ConvertChangeLog.php]: As on many places: Please document more.

File: src/Package/Util/Dumper.php [hunk 45]
```diff
@@ -42,6 +42,7 @@ class Dumper
 {
     /**
      * @param \Pickle\Base\Interfaces\Package $package
```
Comment (Quix0r) [inline src/Package/Util/Dumper.php]: As on many places: Please document more.

File: src/Package/Util/JSON/Dumper.php [hunk 46]
```diff
@@ -43,7 +43,7 @@ class Dumper
 {
     /**
      * @param \Pickle\Base\Interfaces\Package $package
```
Comment (Quix0r) [inline src/Package/Util/JSON/Dumper.php]: As on many places: Please document more.

File: src/Package/Util/JSON/Dumper.php [hunk 47]
```diff
@@ -53,6 +53,8 @@ public function dump(Interfaces\Package $package, $with_version = true)
 
     /**
      * @param \Pickle\Base\Interfaces\Package $package
```
Comment (Quix0r) [inline src/Package/Util/JSON/Dumper.php]: As on many places: Please document more.

**Comment by pierrejoye** [comment]
Thanks (late :/) for your effort!

http > https is fixed in 0.5.0

As per the docblock, happy to add them but it would be better to have a specific PR for those.

**Pull request closed without merging**